### DISCUSSION TITLE
Where do i get the position of the robot when navigating?

### DISCUSSION THREAD

**Discussion by WaitingForAll** [turtlebot/turtlebot4#230] [Q&A] [answered]
For a small task i need the position of the robot while it is navigating. When using the `/odom` topic, i can only get the estimate of the position from the odometry which is of course not the best estimate. Is there a topic which i could subscribe to, to get the position the navigation stack uses? 

I may have mixed up something, but I didn't find anything in the documentation.

Tanks for your help!

Edit: would be the /acml_pose topic a good idea?

**Comment by bmaxdk** [comment, marked as answer]
The robot's position information utilized by the navigation stack generally results from the integration of various sensor inputs. These sensors encompass odometry, and potentially IMU, or other sensors, contingent on the robot's specific configuration. This amalgamated data typically undergoes processing by a localization algorithm such as AMCL, which refines the estimation of the robot's position within the `map` coordinate frame.

Your approach involving the `/amcl_pose` topic is promising. Subscribing to this topic should furnish you with the computed pose of the robot based on the AMCL algorithm. This algorithm considers both odometry and sensor data to align with the robot's position with a known map.
